airbyte-db/db-lib/src/main/resources/sql_functions_and_triggers/config_events.sql

```diff
@@ -0,0 +1,79 @@
+/*
+the goal of this script is to be able to track each change event for a target table. it achieves this by doing a few things:
+1. created a config_event table, where all of the tracked events are stored.
+2. defines a generic psql function that saves a copy of the record AFTER the event is completed. This function can be called on any table that has the following 2 columns: id (uuid, non-nullable) and updated_by (uuid, nullable)
+3. defines a psql function that takes in a table name as an argument and creates a trigger on that table that calls the function defined #2 on insert, updated, or delete.
+
+usage: to add this tracking ability to a table:
+`SELECT create_event_trigger('<table name>');`
+`SELECT create_event_trigger('actor');`
+*/
+
+-- #1 - create config_event table
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+CREATE TYPE operation_type AS ENUM ('insert', 'update', 'delete');
+
+CREATE TABLE IF NOT EXISTS config_event
+(
+    "id"             uuid PRIMARY KEY         DEFAULT uuid_generate_v4(),
+    "operation_type" operation_type  NOT NULL,
+    "config_type"    varchar(256)    NOT NULL,
+    "config_id"      uuid            NOT NULL,
+    "result"         jsonb           NULL,
+    "created_at"     timestamptz(35) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "created_by"     uuid            NULL
+);
+
+CREATE INDEX config_id_idx ON config_event (config_id);
+CREATE INDEX config_type_and_id_idx ON config_event (config_type, config_id);
+
+-- #2 - declare function that saves copy of the record after it is modified.
+-- WARNING: target table must define the following 2 columns: id (uuid, non-nullable) and updated_by (uuid, nullable)
+CREATE OR REPLACE FUNCTION process_config_event() RETURNS TRIGGER AS
+$$
+DECLARE
+    -- functions used in triggers cannot take arguments in the traditional way. they can be passed in via TG_ARGS though.
+    configtype varchar(256);
+BEGIN
+    configtype := TG_ARGV[0];
+    /*
+    Create a row in config_event to reflect the operation performed on the target table. Make use of the special variable TG_OP to work out the operation.
+    */
+    IF (TG_OP = 'DELETE') THEN
+        -- note: for deletes, we cannot track how did the delete.
+        INSERT INTO config_event (operation_type, config_type, config_id, result, created_at, created_by)
+        VALUES ('delete', configtype, OLD.id, NULL, NOW(), NULL);
```

Comment:
This tracking is already a clear improvement from the current state, however, I still wonder, what are our options to track the actor in case of a deletion?